store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/operation/handler/EitherOperationHandler.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.federated.simple.operation.handler;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.gov.gchq.gaffer.operation.Operation;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
+
+/**
+ * Custom handler for operations that could in theory target sub graphs or the
+ * federated store directly.
+ */
+public class EitherOperationHandler<O extends Operation> implements OperationHandler<O> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EitherOperationHandler.class);
+
+    private final OperationHandler<O> standardHandler;
+
+    public EitherOperationHandler(final OperationHandler<O> standardHandler) {
+        this.standardHandler = standardHandler;
+    }
+
+    /**
+     * If graph IDs are in the options the operation will be handled by a
+     * {@link FederatedOperationHandler}, otherwise the original handler will be
+     * used e.g. executed on the federated store directly.
+     */
```

Comment:
Does this still work with default graph ids?

Reply:
Yep as it will end up falling through to just using the default handler which in most cases is the `FederatedOperationHandler`which will just use the default graph IDs. One caveat is that the whole chain is not forwarded though.